drivers/gles3/rasterizer_scene_gles3.cpp

```diff
@@ -1834,15 +1834,14 @@ void RasterizerSceneGLES3::_render_geometry(RenderList::Element *e) {
 
 void RasterizerSceneGLES3::_setup_light(RenderList::Element *e, const Transform &p_view_transform) {
 
-	int omni_indices[16];
+	int maxobj = state.max_forward_lights_per_object;
+	int *omni_indices = (int *)alloca(maxobj * sizeof(int));
```

Comment:
MSVC doesn't seem to like `alloca` here:
```
D:\a\godot\godot\drivers\gles3\rasterizer_scene_gles3.cpp(2219) : error C2220: the following warning is treated as an error
D:\a\godot\godot\drivers\gles3\rasterizer_scene_gles3.cpp(2219) : warning C4750: 'void __cdecl RasterizerSceneGLES3::_setup_light(struct RasterizerSceneGLES3::RenderList::Element * __ptr64,class Transform const & __ptr64) __ptr64': function with _alloca() inlined into a loop
```

Seems like you might have to remove `_FORCE_INLINE_` for it to compile.

BTW, would it make sense to use something smaller than `int` to allocate less memory (e.g. `uint8_t` or `uint16_t`)? Or it doesn't matter?

Reply:
it would use less memory but even with 1000 lights it only uses around 0.005mb, so I don't really think it matters either way

Reply:
`int` is probably fine here as the max size is 1024 * sizeof (int), but in general when using the stack, akien is absolutely right, because the stack can be of very limited size, platform dependent, and how much room you have 'depends'. It absolutely is not the same as general memory.

See e.g. here, suggesting historical figures of 8K being possible on old mobile:
https://stackoverflow.com/questions/1859327/android-stack-size

So 1024 * 8 byte int = 8K, blown stack. Whereas 1024 * 2 byte uint16_t = slightly safer, but still likely to blow such a small stack.

More realistically, modern times I would expect around a 1 meg+, especially on desktop:
https://pspdfkit.com/blog/2019/android-large-memory-requirements/

But this is not just for your function.

In short, alloca is great but make sure you know what you are doing, and beware of anything that might be called recursively. This kind of thing is why the compiler refuses to compile with a FORCE_INLINE.